src/System.Runtime.Serialization.Xml/tests/DataContractSerializer.cs

```diff
@@ -2774,6 +2774,68 @@ public static void DCS_TypeWithDelegate()
         Assert.Equal(value.IntProperty, actual.IntProperty);
     }
 
+    [Fact]
+    public static void DCS_ResolveNameReturnsEmptyNamespace()
+    {
+        EmptyNsContainer instance = new EmptyNsContainer(new EmptyNSAddress());
+        var settings = new DataContractSerializerSettings() { MaxItemsInObjectGraph = int.MaxValue, IgnoreExtensionDataObject = false, PreserveObjectReferences = false };
+        EmptyNsContainer result = SerializeAndDeserialize(instance, null, settings, null, true);
+        bool flag = result.address != null;
+        if (flag)
+        {
+            throw new Exception("Address not null as per CSDMain:73231");
+        }
+        Assert.True(!flag);
+
+        result = (EmptyNsContainer)SingleRoundtripPerEpisode(typeof(EmptyNsContainer), instance, new EmptyNamespaceResolver());
+        if (flag)
+        {
+            throw new Exception("Address not null as per CSDMain:73231");
+        }
+        Assert.True(!flag);
+
+        instance = new EmptyNsContainer(new UknownEmptyNSAddress());
+        settings = new DataContractSerializerSettings() { DataContractResolver = new EmptyNamespaceResolver(), MaxItemsInObjectGraph = int.MaxValue, IgnoreExtensionDataObject = false, PreserveObjectReferences = false };
+        result = SerializeAndDeserialize(instance, null, settings, null, true);
+        if (flag)
+        {
+            throw new Exception("Address not null as per CSDMain:79892");
+        }
+        Assert.True(!flag);
+
+        result = (EmptyNsContainer)SingleRoundtripPerEpisode(typeof(EmptyNsContainer), instance, new EmptyNamespaceResolver());
+        if (flag)
+        {
+            throw new Exception("Address not null as per CSDMain:73231");
+        }
+        Assert.True(!flag);
+    }
+
+    [Fact]
+    public static void DCS_ResolveDatacontractBaseType()
+    {
+        Customer customerInstance = new PreferredCustomerProxy();
+        Type customerBaseType = customerInstance.GetType().BaseType;
+        var settings = new DataContractSerializerSettings() { DataContractResolver = new ProxyDataContractResolver(), MaxItemsInObjectGraph = int.MaxValue, IgnoreExtensionDataObject = false, PreserveObjectReferences = true };
+        object result = SerializeAndDeserialize(customerInstance, null, settings, null, true);
+        Assert.Equal(customerBaseType, result.GetType());
+        result = SingleRoundtripPerEpisode(typeof(Customer), customerInstance, new ProxyDataContractResolver());
```

Comment:
What's the difference between `SingleRoundtripPerEpisode` and `SerializeAndDeserialize`? Do we need to use both of them?